scipy/linalg/basic.py

```diff
@@ -8,7 +8,9 @@
 __all__ = [  # noqa: F822
     'solve', 'solve_triangular', 'solveh_banded', 'solve_banded',
     'solve_toeplitz', 'solve_circulant', 'inv', 'det', 'lstsq',
-    'pinv', 'pinv2', 'pinvh', 'matrix_balance', 'matmul_toeplitz'
+    'pinv', 'pinv2', 'pinvh', 'matrix_balance', 'matmul_toeplitz',
+    'atleast_1d', 'atleast_2d', 'get_flinalg_funcs', 'get_lapack_funcs',
```

Comment:
some functions here, `at_least_nd`s, are NumPy functions and `LinAlgError`, `LinAlgWarning` is imported more than once throughout the linalg files. Is there any script going a bit overzealous here?

Reply:
Not sure what you mean - there is no `at_least_nds` and there's only a single copy of the warning/error objects in this diff here? If you mean that there is one in `basic.py`, one in `decomp.py`, etc. - that is expected. The whole point of this PR is to prevent people getting hard errors when they are importing objects like this from the wrong namespace - so the names of those objects have to be added back to the `__all__` dict of multiple namespaces.

Reply:
I took the shortcut atleast_1d and atleast_2d. But yes I meant multiple files importing the same object that's what I meant.

Reply:
> I took the shortcut atleast_1d and atleast_2d.

Ah makes sense now. Yes they are NumPy functions, and we could try harder to filter those out. However there's not much point here, the easiest and most consistent way of doing things is to set `__all__` to whatever was in `dir(namespace)` before.